src/gl.rs

```diff
@@ -584,6 +597,20 @@ impl<F: 'static + Fn(&Gl, &str, GLenum)> ErrorReactingGl<F> {
     }
 }
 
+/// A wrapper around GL context that times each call and invokes the callback
+/// if the call takes longer than the threshold.
+pub struct ProfilingGl<F> {
+    gl: Rc<Gl>,
```

Comment:
Shouldn't that be a type parameter itself? If `ProfilingGl` is used as a trait object that's twice as many virtual calls, right?

Reply:
I'm not sure how that'd work. This needs to be usable interchangeably with other Rc<Gl> references, similar to ErrorReactingGl.